pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -129,9 +142,24 @@ func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string)
 // older PipelineObject into its v1beta1 equivalent.
 func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object) (v1beta1.PipelineObject, error) {
 	if pipeline, ok := obj.(v1beta1.PipelineObject); ok {
-		pipeline.SetDefaults(ctx)
 		return pipeline, nil
 	}
 
 	return nil, errors.New("resource is not a pipeline")
 }
+
+// verifyResolvedPipeline verifies the resolved pipeline
+func verifyResolvedPipeline(ctx context.Context, pipeline v1beta1.PipelineObject) error {
+	cfg := config.FromContextOrDefaults(ctx)
+	if cfg.FeatureFlags.ResourceVerificationMode != config.SkipResourceVerificationMode && cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
+		if err := trustedresources.VerifyPipeline(ctx, pipeline.Copy()); err != nil {
```

Comment:
nit: why are we passing a copy of the pipeline and not just a pointer to the object?

Reply:
We need to modify the object before verification. e.g. need to extract the signature and delate it from the annotation. I remember I encountered some errors if I don't pass a copy. Let me double check it again.

Reply:
oh it seems no need to pass a copy 😢 I will remove all of them

Reply:
removed. Thanks!